Sources/swift-parser-cli/Commands/PerformanceTest.swift

```diff
@@ -37,12 +41,40 @@ struct PerformanceTest: ParsableCommand {
       .filter { $0.pathExtension == "swift" }
       .map { try Data(contentsOf: $0) }
 
+    var fileTransition: [Data: (SourceFileSyntax, IncrementalParseNodeAffectRangeCollector)] = [:]
+
+    /// The initial parse
+    for file in files {
+      let incrementalParseAffectRangeCollector = self.incrementalParse ? IncrementalParseNodeAffectRangeCollector() : nil
+      file.withUnsafeBytes { buf in
+        let tree = Parser.parse(
+          source: buf.bindMemory(to: UInt8.self),
+          parseNodeAffectRange: incrementalParseAffectRangeCollector
+        )
+
+        if self.incrementalParse == true {
+          fileTransition[file] = (tree, incrementalParseAffectRangeCollector!)
+        }
+      }
+    }
+
     let start = Date()
     let startInstructions = getInstructionsExecuted()
     for _ in 0..<self.iterations {
       for file in files {
+        let parseTransition =
+          incrementalParse ? IncrementalParseTransition(previousTree: fileTransition[file]!.0, edits: ConcurrentEdits(fromSequential: [])) : nil
+        let incrementalParseAffectRangeCollector = incrementalParse ? fileTransition[file]!.1 : nil
         file.withUnsafeBytes { buf in
-          _ = Parser.parse(source: buf.bindMemory(to: UInt8.self))
+          let tree = Parser.parse(
+            source: buf.bindMemory(to: UInt8.self),
+            parseNodeAffectRange: incrementalParseAffectRangeCollector,
+            parseTransition: parseTransition
+          )
+
+          if self.incrementalParse == true {
+            fileTransition[file] = (tree, incrementalParseAffectRangeCollector!)
+          }
```

Comment:
We don’t need this, right? We could just perform `self.iterations` incremental parses based on the tree generation in the initial parse loop.

Reply:
Ah, yeah.